src/main/java/frc/robot/subsystems/Indexer.java

```diff
@@ -0,0 +1,335 @@
+            }
+
+            @Override
+            public boolean isFinished() {
+                return topPierreIR.getAsBoolean();
+                // values of .8 when open, 2.4 when closed
+
+            }
+
+            @Override
+            public void execute() {
+                pierreMotor.set(pierreIndexSpeed);
+
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                pierreMotor.set(0);
+               
+            }
+
+        };
+
+    }
+    //this will bring the ball to the top
+    
+    public CommandBase unLoadBall() {
+        return new CommandBase() {
+
+            {
+                addRequirements(Indexer.this);
+            }
+
+            @Override
+            public boolean isFinished() {
+                return !topPierreIR.getAsBoolean();
+                // values of .8 when open, 2.4 when closed
+
+            }
+
+            @Override
+            public void execute() {
+                pierreMotor.set(pierreIndexSpeed);
+
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                pierreMotor.set(0);
+                ballCounting--;
+                SmartDashboard.putNumber("Ball Count", ballCounting);
+
+            }
+
+        };
+
+    }
+    //this will bring the ball to the shooter
+
+    public CommandBase runToClear() {
+        return new CommandBase() {
+
+            {
+                addRequirements(Indexer.this);
+            }
+
+            @Override
+            public void initialize() {
+                if (bottomPierreIR.getAsBoolean()) {
+                    pierreMotor.set(pierreIndexSpeed);
+                }
+            }
+
+            @Override
+            public boolean isFinished() {
+                return !bottomPierreIR.getAsBoolean();
+                // values of .8 when open, 2.4 when closed
+
+            }
+
+            @Override
+            public void execute() {
+                // pierreMotor.set(0);
+
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                pierreMotor.set(0);
+                ballCounting++;
+                SmartDashboard.putNumber("Ball Count", ballCounting);
+
+            }
+
+        };
+
+    }
+    // this will make space for another ball
+    public CommandBase stopWhenBallsAtTop() {
+        return new CommandBase() {
+
+
+            @Override
+            public void initialize() {
+
+            }
+
+            @Override
+            public boolean isFinished() {
+                return topPierreIR.getAsBoolean();
+                // values of .8 when open, 2.4 when closed
+
+            }
+
+            @Override
+            public void execute() {
+
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                pierreMotor.set(0);
+                singulator.set(0);
```

Comment:
These are unnecessary as the running command should also stop the motor in its own `end`